src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -846,14 +842,9 @@ private static String sanitizeValueForExceptionMessage(Object value) {
    * UTF-16 surrogate, for example.
    */
   private static void verifyValidUtf8(String input, String columnName, String dataType) {
-    CharsetEncoder charsetEncoder =
-        StandardCharsets.UTF_8
-            .newEncoder()
-            .onMalformedInput(CodingErrorAction.REPORT)
-            .onUnmappableCharacter(CodingErrorAction.REPORT);
-    try {
-      charsetEncoder.encode(CharBuffer.wrap(input));
-    } catch (CharacterCodingException e) {
+    String roundTripStr =
```

Comment:
Interesting idea :)
I guess the only difference is how informative `CharsetEncoder`'s exception could have been.
This new approach just detects that something is wrong.
Should we parameterise this change for debugging purposes?

Reply:
We output the data so it should sufficient for us to debug?